engine/language-server/src/main/scala/org/enso/languageserver/runtime/ContextRegistry.scala

```diff
@@ -110,11 +113,11 @@ final class ContextRegistry(
           .foreach(_ ! update)
 
       case update: Api.ExecutionFailed =>
-        sampler.stop(6.seconds)(context.dispatcher)
+        sampler.stop(profilingTimeout)(context.dispatcher)
         store.getListener(update.contextId).foreach(_ ! update)
 
       case update: Api.ExecutionComplete =>
-        sampler.stop(6.seconds)(context.dispatcher)
+        sampler.stop(profilingTimeout)(context.dispatcher)
```

Comment:
The sampler is started at the beginning and it runs until the timeout is reached. Then the file is written to disk and sampling stops. Is that what we want, @kazcw , @MichaelMauderer ? If the IDE specifies small timeout, recording may stop sooner than user performs the action. The IDE can specify huge timeout, but then: when the file is going to be saved to disk?

Reply:
In the frontend we usually save the profile just before exit. When the frontend exits, it causes the backend to shut down too. Can the backend save the profile before exiting?

Reply:
Following could work:
- when no timeout is specified, it defaults to infinity
- when the engine language server receives "exit request", it `stop`s profiling and dumps the sampler data

We may need further tweaks in the future, but for now this shall be good enough for the _batch mode_ purposes. I think.

Reply:
I updated the logic

Reply:
If this is implemented, please also update the PR description.